src/mongo_logger.ts

```diff
@@ -732,12 +740,40 @@ export class MongoLogger {
     this.componentSeverities = options.componentSeverities;
     this.maxDocumentLength = options.maxDocumentLength;
     this.logDestination = options.logDestination;
+    this.pendingLog = undefined;
   }
 
   willLog(severity: SeverityLevel, component: MongoLoggableComponent): boolean {
     return compareSeverity(severity, this.componentSeverities[component]) <= 0;
   }
 
+  turnOffSeverities() {
+    for (const key of Object.keys(MongoLoggableComponent)) {
+      this.componentSeverities[key as MongoLoggableComponent] = SeverityLevel.OFF;
+    }
+  }
+
+  private logWriteFailureHandler(error: Error) {
+    try {
+      const tempLogDestination = createStdioLogger(process.stderr);
+      tempLogDestination.write({
+        t: new Date(),
+        c: MongoLoggableComponent.CLIENT,
+        s: SeverityLevel.ERROR,
+        message: `User input for mongodbLogPath is now invalid. Now logging to stderr.`,
+        error: error.message
+      });
+      this.logDestination = tempLogDestination;
+    } catch (e) {
+      this.turnOffSeverities();
```

Comment:
Should we be turning off logging for any error?

Reply:
I thought we wanted to stop logging after emitting this warning not to continue using stderr. It is strange to me that we would start logging to that destination for the remainder of the life of the process, the user probably has nothing set up to capture from there otherwise they would've configured stderr to be captured.

Reply:
We determined that this was our course of action through this [slack thread in drivers-1204](https://mongodb.slack.com/archives/G015EB3QHED/p1704916869488999), which points to this [spec section](https://github.com/mongodb/specifications/blob/master/source/logging/logging.rst#configurable-log-destination).

Reply:
Following that thread, I think single message to stderr and then ceasing future logs is the approach we want to go with.

The linked spec refers to putting a message on stderr when misconfigured, here we're concerned with a "runtime" during operation error.

Reply:
Sounds good, I'll configure it to turn off logging once an error with the stream is encountered and the warning message has been logged.